src/opentracing/span.js

```diff
@@ -41,7 +41,7 @@ class DatadogSpan extends Span {
         parentId: parent.spanId,
         sampled: parent.sampled,
         baggageItems: Object.assign({}, parent.baggageItems),
-        trace: parent.trace
+        trace: parent.trace.started.length !== parent.trace.finished.length ? parent.trace : null
```

Comment:
Should there be a method on `SpanContext` to provide this information?

If the internals of `SpanContext` were to change then this would break. Maybe something like `parent.isFinished()` or `parent.isFlushed()`?

Reply:
I was trying to keep SpanContext as a simple bag to hold information without logic, but I agree this logic should be encapsulated.

However, the span has to know this information since it's the span that fills both of these properties with data. Maybe the span could detect this once when adding itself to the `finished` array and set a `finished` property to true? Then any other parts of the code would not have to duplicate this logic, for example on this line or in the writer.

Reply:
I think having the method on `SpanContext` is fine, but I understand the reasoning of wanting to keep it logic-free.

Since the `Span` is in charge of manipulating the `SpanContext` I also think it's fine to perform and keep the logic in `Span`, just including/setting a `finished` property on the `SpanContext`. Since `Span` should be the only thing acting upon a `SpanContext` this seems like the better approach.

Reply:
There are quite a few places that use these values directly in tests and benchmarks, so I think this should be done in another PR.

Opened an issue to track it: https://github.com/DataDog/dd-trace-js/issues/278